src/nvim/os/pty_process_unix.c

```diff
@@ -73,6 +73,12 @@ int pty_process_spawn(PtyProcess *ptyproc)
     goto error;
   }
 
+  // Other jobs and providers shouldn't get an copy of this file descriptor.
+  if (os_set_cloexec(master) == -1) {
```

Comment:
Good point.

I'll change `os_set_cloexec()` instead. Though the linux and bsd man pages say `F_SETFD` will return `0` on success and `-1` on error, the [POSIX spec](http://pubs.opengroup.org/onlinepubs/9699919799/) allows any value other than `-1` to be returned on error, and so does the [osx man page](http://www.manpages.info/macosx/fcntl.2.html).